lib/core/auth_service.dart

```diff
@@ -39,4 +41,26 @@ class AuthService with Store {
 
     return decodedPin == pin;
   }
+
+  void saveLastAuthTime(){
+    int timestamp = DateTime.now().millisecondsSinceEpoch;
+    sharedPreferences.setInt(PreferencesKey.lastAuthTimeMilliseconds, timestamp);
+  }
+
+  bool requireAuth(){  
+      final timestamp = sharedPreferences.getInt(PreferencesKey.lastAuthTimeMilliseconds);
+      final duration =  _durationToRequireAuth(timestamp ?? 0);
+      final requiredPinInterval = getIt.get<SettingsStore>().pinTimeOutDuration;
+     
+      return duration >= requiredPinInterval.value;
+    }
+
+  int _durationToRequireAuth(int timestamp){
+
+      DateTime before = DateTime.fromMillisecondsSinceEpoch(timestamp);
+      DateTime now = DateTime.now();
+      Duration timeDifference = now.difference(before);
+
+      return timeDifference.inMinutes; 
+  }
```

Comment:
Why here content of functions/methods have different left offset ? Please add space between ending of list functions/methods arguments and curly braces like: `void abs(A a, B b)<SPACE>{`.